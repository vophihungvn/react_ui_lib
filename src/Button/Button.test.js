import React from 'react';
import { shallow } from 'enzyme';

import AccountBalance from '.';

describe('AccountBalance Snapshot', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <AccountBalance title={'Account balance'} amount={1000} />
    );
  });

  it('renders without crashing', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
