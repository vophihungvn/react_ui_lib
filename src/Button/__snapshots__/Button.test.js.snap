// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountBalance Snapshot matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    amount={1000}
    title="Account balance"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bg": "blue",
      "border": 0,
      "borderRadius": 2,
      "children": "Account balance",
      "color": "white",
      "fontSize": 1,
      "fontWeight": "bold",
      "is": "button",
      "lineHeight": 1.1428571428571428,
      "m": 0,
      "px": 3,
      "py": 2,
    },
    "ref": null,
    "rendered": "Account balance",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bg": "blue",
        "border": 0,
        "borderRadius": 2,
        "children": "Account balance",
        "color": "white",
        "fontSize": 1,
        "fontWeight": "bold",
        "is": "button",
        "lineHeight": 1.1428571428571428,
        "m": 0,
        "px": 3,
        "py": 2,
      },
      "ref": null,
      "rendered": "Account balance",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
